src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,73 @@
+
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] supplyBuy = calculateStatistic(fromFileName);
+        writeFile(toFileName, createReport(supplyBuy[0], supplyBuy[1]));
+    }
 
+    private int[] calculateStatistic(String fromFileName) {
+        int supply = 0;
+        int buy = 0;
+        String[] stringFile = readFile(fromFileName);
+        for (int i = 0; i < stringFile.length; i++) {
+            String[] splitData = stringFile[i].split(",");
+            if (splitData[0].equals("supply")) {
+                supply += convertStringDataToInteger(splitData[1]);
+            }
+            if (splitData[0].equals("buy")) {
+                buy += convertStringDataToInteger(splitData[1]);
+            }
```

Comment:
lets add constants for indexes to make your code easier to read
`Integer.parseInt(splitData[AMOUNT_INDEX])` - should work perfectly